platform/ios/src/MGLMapView.mm

```diff
@@ -1128,9 +1128,68 @@ - (void)wakeGL:(__unused NSNotification *)notification
         [self validateLocationServices];
 
         [MGLMapboxEvents pushEvent:MGLEventTypeMapLoad withAttributes:@{}];
+        
+        [self updateLocalization];
+    }
+}
+
+- (void)updateLocalization
+{
+    NSString *mapboxLanguage = [[NSUserDefaults standardUserDefaults] stringForKey:@"MGLMapboxLanguage"];
+    NSString *language = [[[NSBundle mainBundle] preferredLocalizations] firstObject];
+    NSString *preferredLanguageCode = [self preferredLanguageCode];
+    
+    if (![mapboxLanguage isEqualToString:language] && ![mapboxLanguage isEqualToString:preferredLanguageCode]) {
+        mapboxLanguage = preferredLanguageCode;
+        [[NSUserDefaults standardUserDefaults] setObject:mapboxLanguage forKey:@"MGLMapboxLanguage"];
+        [self setLabelLanguage:mapboxLanguage];
+        [[NSNotificationCenter defaultCenter] postNotificationName:NSCurrentLocaleDidChangeNotification object:nil];
+    }
+}
+
+- (void)setLabelLanguage:(NSString *)language
+{
+    NSString *preferredNameToken = [NSString stringWithFormat:@"{name_%@}", language];
+    
+    for (MGLSymbolStyleLayer *layer in [[self style] layers]) {
+        if (![layer isKindOfClass:[MGLSymbolStyleLayer class]]) {
+            continue;
+        }
+        
+        if ([layer.text isKindOfClass:[MGLConstantStyleValue class]]) {
+            MGLConstantStyleValue *label = (MGLConstantStyleValue<NSString *> *)layer.text;
+            if ([label.rawValue hasPrefix:@"{name"]) {
+                layer.text = [MGLStyleValue valueWithRawValue:preferredNameToken];
+            }
+        } else if ([layer.text isKindOfClass:[MGLCameraStyleFunction class]]) {
+            MGLCameraStyleFunction *function = (MGLCameraStyleFunction<NSString *> *)layer.text;
+            NSMutableDictionary *stops = function.stops.mutableCopy;
+            [stops enumerateKeysAndObjectsUsingBlock:^(NSNumber *zoomLevel, MGLConstantStyleValue<NSString *> *stop, BOOL *done) {
+                if ([stop.rawValue hasPrefix:@"{name"]) {
+                    stops[zoomLevel] = [MGLStyleValue<NSString *> valueWithRawValue:preferredNameToken];
+                }
+            }];
+            function.stops = stops;
+            layer.text = function;
+        }
     }
 }
 
+- (NSString *)preferredLanguageCode {
+    // Languages supported by Mapbox Streets v10.
+    NSSet *supportedLanguages = [NSSet setWithObjects:@"en", @"es", @"fr", @"de", @"ru", @"zh", nil];
```

Comment:
This is missing `ar`, `pr`, and `zh-Hans`. See [the docs for `mapbox-streets-v7`](https://www.mapbox.com/vector-tiles/mapbox-streets-v7/#overview) and [this help doc](https://www.mapbox.com/help/change-language/) for more information.